HW10/classes.js

```diff
@@ -0,0 +1,171 @@
+/** Завдання створити класи:
+ *    фігура:
+ *      координати:
+ *      х, у
+ *
+ *    багатокутник -> фігура:
+ *      вершини[координати],
+ *      периметр()
+ *
+ *    прямокутник -> багатокутник:
+ *      довжина,
+ *      висота,
+ *      площа(),
+ *      периметр()
+ *
+ *    квадрат -> прямокутник:
+ *      довжинаСторони,
+ *      площа(),
+ *      периметр()
+ *
+ *    круг -> фігура:
+ *      радіус,
+ *      площа(),
+ *      периметр()
+ */
+
+class Shape {
+  /**
+   * @param {array} center
+   */
+  constructor(center) {
+    this.center = center;
+  }
+}
+
+/* багатокутник -> фігура:
+ *      вершини[координати],
+ *      периметр() */
+class Polygon extends Shape {
+  /**
+  * @param {array} center
+  * @param {object} points
```

Comment:
object - це дуже узагальнена структура данних
```suggestion
  * @param {Point[]} points
```